src/utils/Meta.js

```diff
@@ -1,60 +1,88 @@
+ */
+const getMeta = (arg, key) => {
+  let _meta = {}
+
+  // literal
+  if (isMeta(arg)) _meta = arg
+
+  // from prop
+  else if (isMeta(_.get(arg, '_meta'))) _meta = arg._meta
+
+  // from class
+  else if (isMeta(_.get(arg, 'constructor._meta'))) _meta = arg.constructor._meta
+
+  // default values
+  return key ? _meta[key] : _meta
+}
 
 /**
  * Component meta information.  Used to declaratively classify and identify components.
  * @type {{}}
  */
 const META = {
-  library: {
-    semanticUI: 'Semantic UI',
-    stardust: 'Stardust',
-  },
-
-  type: {
-    addon: 'addon',
-    global: 'global',
-    collection: 'collection',
-    element: 'element',
-    view: 'view',
-    module: 'module',
-  },
+  library: LIBRARIES,
+  type: TYPES,
 
   // library
-  isSemanticUI: ({ _meta }) => _meta.library === META.library.semanticUI,
-  isStardust: ({ _meta }) => _meta.library === META.library.stardust,
+  isSemanticUI: (arg) => getMeta(arg, 'library') === META.library.semanticUI,
+  isStardust: (arg) => getMeta(arg, 'library') === META.library.stardust,
 
   // type
-  isAddon: ({ _meta }) => _meta.type === META.type.addon,
-  isGlobal: ({ _meta }) => _meta.type === META.type.global,
-  isCollection: ({ _meta }) => _meta.type === META.type.collection,
-  isElement: ({ _meta }) => _meta.type === META.type.element,
-  isView: ({ _meta }) => _meta.type === META.type.view,
-  isModule: ({ _meta }) => _meta.type === META.type.module,
-  isType: ({ _meta }, type) => _meta.type === type,
+  isAddon: (arg) => getMeta(arg, 'type') === META.type.addon,
+  isGlobal: (arg) => getMeta(arg, 'type') === META.type.global,
+  isCollection: (arg) => getMeta(arg, 'type') === META.type.collection,
+  isElement: (arg) => getMeta(arg, 'type') === META.type.element,
+  isView: (arg) => getMeta(arg, 'type') === META.type.view,
+  isModule: (arg) => getMeta(arg, 'type') === META.type.module,
+  isType: (arg, type) => getMeta(arg, 'type') === type,
 
   // parent
-  isParent: (component) => (
-    // has no 'parent' or 'parent' is self
-    !_.has(component, '_meta.parent') || _.get(component, '_meta.parent') === component._meta.name
-  ),
-  isChild: component => !META.isParent(component),
+  isParent: (arg) => !getMeta(arg, 'parent'),
+  isChild: arg => !!getMeta(arg, 'parent'),
 
   // other
-  /**
-   * Components whose constructor begins with an "_" are private.
-   * This helper handles strings, classes, and instances and returns true if the
-   * name or class definition begins with an "_".
-   * @param {string|class|object} component A class, an instance, or constructor name.
-   * @returns {boolean}
-   */
-  isPrivate: component => {
-    // handle component names from a string, a class, or an instance
-    const name = _.isString(component) && component
-      || _.get(component, '_meta.name')
-      || _.get(component, '.constructor._meta.name')
-
-    return _.startsWith(name, '_')
-  },
+  isPrivate: arg => _.startsWith(getMeta(arg, 'name'), '_'),
```

Comment:
Both the above utils facilitate our `is*` methods here.  We can now ask if a class, instance, or object `is*`.  This is immensely helpful in automated common tests and doc site operations.